unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Avatar.cs

```diff
@@ -57,37 +59,46 @@ public class Avatar : IAvatar
         /// <param name="ct"></param>
         public async UniTask Load(List<string> wearablesIds, List<string> emotesIds, AvatarSettings settings, CancellationToken ct = default)
         {
-            disposeCts ??= new CancellationTokenSource();
-
             status = IAvatar.Status.Idle;
-            CancellationToken linkedCt = CancellationTokenSource.CreateLinkedTokenSource(ct, disposeCts.Token).Token;
 
-            linkedCt.ThrowIfCancellationRequested();
+            loadCancellationToken = loadCancellationToken.SafeRestart();
+            CancellationToken linkedCt = CancellationTokenSource.CreateLinkedTokenSource(ct, loadCancellationToken.Token).Token;
 
             try
             {
                 await LoadTry(wearablesIds, emotesIds, settings, linkedCt);
             }
             catch (OperationCanceledException)
             {
-                Dispose();
+                // Cancel any ongoing process except the current loadCancellationToken
+                // since it was provoking a double cancellation thus inconsistencies in the flow
+                status = IAvatar.Status.Idle;
+                avatarCurator?.Dispose();
+                loader?.Dispose();
+                visibility?.Dispose();
+                lod?.Dispose();
+                gpuSkinningThrottlerService?.Unregister(gpuSkinning);
```

Comment:
I would suggest extracting this piece of code to a method, since its repeated in three places in this class.